qt/place_page_dialog.cpp

```diff
@@ -84,6 +84,37 @@ PlacePageDialog::PlacePageDialog(QWidget * parent, place_page::Info const & info
     label->setTextInteractionFlags(Qt::TextSelectableByMouse);
     grid->addWidget(label, row++, 1);
   }
+  {
+    auto busRoutes = info.GetBusRoutes();
```

Comment:
With a single method for three types of routes you'll be able to use a loop here instead of three almost identical blocks of code.